src/common/shm/ngx_wasm_shm.c

```diff
@@ -54,6 +54,13 @@ ngx_wasm_shm_init(ngx_cycle_t *cycle)
             }
             dd("queue init done");
             break;
+        case NGX_WASM_SHM_TYPE_METRICS:
+            rc = ngx_wasm_shm_kv_init(shm);
```

Comment:
This is also called in `ngx_wa_metrics_init` - can we only keep one invocation? Or perhaps better, invoke `ngx_wa_metrics_init` here instead of in `ngx_wasmx.c`?

Reply:
Actually, this PR was moving `ngx_wasm_shm_kv_init` out of `ngx_wa_metrics_init` to here. 

I agree that it makes more sense to have `ngx_wa_metrics_init` in here, rather than in `ngx_wasmx.c`, and put `ngx_wasm_shm_kv_init` back to `ngx_wa_metric_init`, though.

Reply:
I missed the line that removed the old `ngx_wasm_shm_kv_init`. But yes, let's do the other thing which I think is more explicit to readers of this `ngx_wasm_shm_init` switch case.